sql/core/src/test/scala/org/apache/spark/sql/DataFrameComplexTypeSuite.scala

```diff
@@ -58,4 +58,38 @@ class DataFrameComplexTypeSuite extends QueryTest with SharedSQLContext {
     val nullIntRow = df.selectExpr("i[1]").collect()(0)
     assert(nullIntRow == org.apache.spark.sql.Row(null))
   }
+
+  test("SPARK-15285 Generated SpecificSafeProjection.apply method grows beyond 64KB") {
+    val ds100_5 = Seq(S100_5()).toDS()
+    ds100_5.show
```

Comment:
hmm, this is weird, the main logic in `show` is calling `collect` to get the data, can you double check it? and how about `ds.rdd.collect`?